kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/batch/TableWriter.java

```diff
@@ -116,10 +117,10 @@ public void run() {
     onFinish.accept(rows);
   }
 
-  private static int getNewBatchSize(int currentBatchSize) {
+  private static int getNewBatchSize(int currentBatchSize, Throwable err) {
     if (currentBatchSize == 1) {
       // todo correct exception type?
-      throw new ConnectException("Attempted to reduce batch size below 1.");
+      throw new BigQueryConnectException("Attempted to reduce batch size below 1.", err);
```

Comment:
I think this can be improved a bit but if this is sufficient for error mapping and log debugging for cloud on-call then we can leave it as-is for now.

Reply:
Do you mean the error message or the exception type?

Reply:
The message... don't think the `Attempted to reduce batch size below 1` part is really helpful here; it's an internal detail that is meaningless to 99% of users and a lot of engineers too.

Reply:
I agree. We can revisit this later.